app/src/main/kotlin/com/antosik/benchproject/app/popular/movies/viewModel/PopularMoviesViewModel.kt

```diff
@@ -39,6 +41,14 @@ class PopularMoviesViewModel @Inject constructor(
         loadPopularMovies()
     }
 
+    fun updatePopularMovie(movie: Movie) {
+        viewModelScope.launch {
+            val movieModel = movie.copy(isFavorite = !movie.isFavorite).toDomain()
```

Comment:
It should be done in UseCase

Reply:
Because it is business logic? To be honest from first time when I created the domain layer I wondered when I should add some logic to this place and I tried figure out what I can add to this.

Reply:
So it could look like: 
```
    suspend operator fun invoke(movieModel: MovieModel) {
        repo.updatePopularMovie(movieModel.copy(isFavorite = !movieModel.isFavorite))
    }
```

Reply:
Yes, all business logic should be placed in domain layer, app layer should be "stupid simple" 😄